Objects/codeobject.c

```diff
@@ -868,6 +868,8 @@ _PyCode_SetExtra(PyObject *code, Py_ssize_t index, void *extra)
         }
         co_extra = (_PyCodeObjectExtra *) o->co_extra;
 
+        co_extra->ce_size = 0;
+
         co_extra->ce_extras = PyMem_Malloc(
             tstate->co_extra_user_count * sizeof(void*));
         if (co_extra->ce_extras == NULL) {
```

Comment:
Would not be better to deallocate `o->co_extra` and set it to `NULL` rather of setting `co_extra->ce_size` to `0`?